framework/src/seclink/seclink.c

```diff
@@ -174,6 +174,48 @@ int sl_deinit(sl_ctx hnd)
 	return 0;
 }
 
+#ifdef CONFIG_ARM_TRUSTZONE
+int sl_allocate(sl_ctx hnd, uint32_t ssize)
+{
+	SL_ENTER;
+	if (!hnd || ((struct _seclink_s_ *)hnd)->fd <= 0) {
+		SL_ERR(-1);
+		return SECLINK_ERROR;
+	}
+
+	struct _seclink_s_ *sl = (struct _seclink_s_ *)hnd;
+	struct seclink_comm_info info = {.priv = NULL, .s_size = ssize, .s_context = NULL};
+	struct seclink_req req = {.req_type.comm = &info, 0};
+
+	SL_CALL(sl, SECLINKIOC_ALLOC, req);
+
+	if (req.res != HAL_SUCCESS) {
+		SL_LOG("fail to allocate");
```

Comment:
we don't return error in this case?

Reply:
sangam, overall in this file, error reporting is not proper. most of the apis in this file does not verify the return value. Moreover, a secure context allocation failure will not return here, it would result in Secure Fault